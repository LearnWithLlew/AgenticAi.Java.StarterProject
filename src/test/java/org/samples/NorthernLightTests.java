package org.samples;

import org.approvaltests.Approvals;
import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.Test;

public class NorthernLightTests {
    @Test
    public void canFetchNoaaAuroraAlerts() throws Exception {
        String city = "Fairbanks";
        double latitude = 64.8378;
        double longitude = -147.7164;
        String result = NoaaAuroraService.getAuroraAlert(city, latitude, longitude);
        JsonApprovals.verifyJson(result);
    }
}
