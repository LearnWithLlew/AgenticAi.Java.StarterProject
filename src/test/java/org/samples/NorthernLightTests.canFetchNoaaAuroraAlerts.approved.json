{
  "date": "2025-07-25T01:39:42+00:00",
  "ace": {
    "date": "2025-07-25T01:39:42+00:00",
    "bz": "0",
    "density": "0.48",
    "speed": "573.1",
    "kp1hour": "2.33",
    "kp4hour": "2.33",
    "kp": "2.33",
    "colour": {
      "bz": "green",
      "density": "green",
      "speed": "orange",
      "kp1hour": "green",
      "kp4hour": "green",
      "kp": "green"
    }
  },
  "weather": false,
  "threeday": {
    "date": "2025-07-25T01:39:42+00:00",
    "dates": [
      "2025-07-25T00:00:00+00:00",
      "2025-07-26T00:00:00+00:00",
      "2025-07-27T00:00:00+00:00"
    ],
    "values": [
      [
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": true,
          "start": "2025-07-25T00:00:00+00:00",
          "end": "2025-07-25T03:00:00+00:00",
          "value": "3.67",
          "colour": "yellow"
        },
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": false,
          "start": "2025-07-25T03:00:00+00:00",
          "end": "2025-07-25T06:00:00+00:00",
          "value": "3.00",
          "colour": "yellow"
        },
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": false,
          "start": "2025-07-25T06:00:00+00:00",
          "end": "2025-07-25T09:00:00+00:00",
          "value": "2.67",
          "colour": "green"
        },
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": false,
          "start": "2025-07-25T09:00:00+00:00",
          "end": "2025-07-25T12:00:00+00:00",
          "value": "2.33",
          "colour": "green"
        },
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": false,
          "start": "2025-07-25T12:00:00+00:00",
          "end": "2025-07-25T15:00:00+00:00",
          "value": "2.00",
          "colour": "green"
        },
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": false,
          "start": "2025-07-25T15:00:00+00:00",
          "end": "2025-07-25T18:00:00+00:00",
          "value": "2.33",
          "colour": "green"
        },
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": false,
          "start": "2025-07-25T18:00:00+00:00",
          "end": "2025-07-25T21:00:00+00:00",
          "value": "2.33",
          "colour": "green"
        },
        {
          "date": "2025-07-25T00:00:00+00:00",
          "now": false,
          "start": "2025-07-25T21:00:00+00:00",
          "end": "2025-07-26T00:00:00+00:00",
          "value": "2.67",
          "colour": "green"
        }
      ],
      [
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": true,
          "start": "2025-07-26T00:00:00+00:00",
          "end": "2025-07-26T03:00:00+00:00",
          "value": "2.33",
          "colour": "green"
        },
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": false,
          "start": "2025-07-26T03:00:00+00:00",
          "end": "2025-07-26T06:00:00+00:00",
          "value": "1.67",
          "colour": "green"
        },
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": false,
          "start": "2025-07-26T06:00:00+00:00",
          "end": "2025-07-26T09:00:00+00:00",
          "value": "1.33",
          "colour": "green"
        },
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": false,
          "start": "2025-07-26T09:00:00+00:00",
          "end": "2025-07-26T12:00:00+00:00",
          "value": "1.33",
          "colour": "green"
        },
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": false,
          "start": "2025-07-26T12:00:00+00:00",
          "end": "2025-07-26T15:00:00+00:00",
          "value": "1.00",
          "colour": "green"
        },
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": false,
          "start": "2025-07-26T15:00:00+00:00",
          "end": "2025-07-26T18:00:00+00:00",
          "value": "1.33",
          "colour": "green"
        },
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": false,
          "start": "2025-07-26T18:00:00+00:00",
          "end": "2025-07-26T21:00:00+00:00",
          "value": "1.33",
          "colour": "green"
        },
        {
          "date": "2025-07-26T00:00:00+00:00",
          "now": false,
          "start": "2025-07-26T21:00:00+00:00",
          "end": "2025-07-27T00:00:00+00:00",
          "value": "1.67",
          "colour": "green"
        }
      ],
      [
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": true,
          "start": "2025-07-27T00:00:00+00:00",
          "end": "2025-07-27T03:00:00+00:00",
          "value": "2.00",
          "colour": "green"
        },
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": false,
          "start": "2025-07-27T03:00:00+00:00",
          "end": "2025-07-27T06:00:00+00:00",
          "value": "2.00",
          "colour": "green"
        },
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": false,
          "start": "2025-07-27T06:00:00+00:00",
          "end": "2025-07-27T09:00:00+00:00",
          "value": "3.67",
          "colour": "yellow"
        },
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": false,
          "start": "2025-07-27T09:00:00+00:00",
          "end": "2025-07-27T12:00:00+00:00",
          "value": "5.00 (G1)",
          "colour": "orange"
        },
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": false,
          "start": "2025-07-27T12:00:00+00:00",
          "end": "2025-07-27T15:00:00+00:00",
          "value": "4.00",
          "colour": "yellow"
        },
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": false,
          "start": "2025-07-27T15:00:00+00:00",
          "end": "2025-07-27T18:00:00+00:00",
          "value": "3.33",
          "colour": "yellow"
        },
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": false,
          "start": "2025-07-27T18:00:00+00:00",
          "end": "2025-07-27T21:00:00+00:00",
          "value": "3.00",
          "colour": "yellow"
        },
        {
          "date": "2025-07-27T00:00:00+00:00",
          "now": false,
          "start": "2025-07-27T21:00:00+00:00",
          "end": "2025-07-28T00:00:00+00:00",
          "value": "3.33",
          "colour": "yellow"
        }
      ]
    ]
  },
  "probability": {
    "date": "2025-07-25T01:39:42+00:00",
    "calculated": {
      "lat": 73.125,
      "long": -156.445,
      "value": 5,
      "colour": "green"
    },
    "colour": "green",
    "lat": "64.8378",
    "long": "-147.7164",
    "value": 0,
    "highest": {
      "date": "2025-07-25T01:39:42+00:00",
      "colour": "orange",
      "lat": 54.8438,
      "long": 80.5078,
      "value": 53
    }
  },
  "message": [
    "Due to changes to data from the Space Weather Prediction Centre, the 1 hour and 4 hour Kp forecasts are unavailable and have been replaced with the current Kp. This issue will be resolved when alternate data is available"
  ]
}